addon/mixins/legend.js

```diff
@@ -226,6 +245,10 @@ export default Ember.Mixin.create({
     }
   }).volatile(),
 
+  // Create a list of all the legend Items, icon for each legend item and corresponding labels
+  // Calculate the number of legend item rows and items in each. Each time width should be bounded by min and max legend item width.
+  // Calculate the label width for each legend row that minimizes truncation. 
+  // And then apply legendItemAttrs to apply posisioning transforms.
   drawLegend: function() {
```

Comment:
This whole function is really big, and should probably be broken into helper functions.